src/vips/Image_methods.lua

```diff
@@ -388,14 +387,21 @@ function Image_method:write_to_memory()
     local psize = ffi.new(gvalue.psize_typeof, 1)
     local vips_memory = vips_lib.vips_image_write_to_memory(self.vimage, psize)
     local size = psize[0]
-    -- FIXME can we avoid the copy somehow?
+
     local lua_memory = ffi.new(gvalue.mem_typeof, size)
     ffi.copy(lua_memory, vips_memory, size)
     glib_lib.g_free(vips_memory)
 
     return lua_memory
 end
 
+function Image_method:write_to_memory_nocopy()
```

Comment:
Sorry, thinking about this again, how about calling them `write_to_memory_ptr` and `new_from_memory_ptr`?

It makes it clear that they are a pair of methods dealing with non-FFI-managed memory. The connection is less obvious if we use a different name.